backend/src/main/java/com/gigigenie/domain/chat/service/ChatService.java

```diff
@@ -29,51 +30,95 @@ public class ChatService {
 
-        // FastAPI /chat 호출
-        String answer = ragWebClient.post()
+        // FastAPI /chat 호출  타입수정 필요해 옵입니다 response 수정
+        // webClient 예외처리부분도 추가해주세요
+        String botResponse = ragWebClient.post()
             .uri("/chat")
-            .bodyValue(body)
+            .bodyValue(fastApiRequest)
             .retrieve()
             .bodyToMono(String.class)
             .doOnNext(res -> log.info("응답: {}", res))
             .switchIfEmpty(Mono.error(new IllegalStateException("응답이 비어있음")))
             .block();
 
-        if (!hasSession) { // 세션 X
-            List<ChatMessage> newHistory = new ArrayList<>();
-            newHistory.add(new ChatMessage("user", request.getQuestion()));
-            newHistory.add(new ChatMessage("assistant", answer));
-            chatStore.save(member.getMemberId(), product.getId(), newHistory);
-        } else { // 세션 O
-            List<ChatMessage> updated = new ArrayList<>(stored);
-            updated.add(new ChatMessage("user", request.getQuestion()));
-            updated.add(new ChatMessage("assistant", answer));
-            chatStore.save(member.getMemberId(), product.getId(), updated);
+        ChatMessage botMessage = new ChatMessage("bot", botResponse);
+        history.add(botMessage);
+        redisChatService.save(memberId, productId, history);
+
+        return history;
+    }
+
+    // 대화 종료
+    public void endChat(Long productId, Authentication authentication, boolean skipSave) {
+        // 1) 식별자 조회
+        Member member = findMember(authentication);
+        Product product = findProduct(productId);
+
+        // 2) 대화 내역 불러오기
+        List<ChatMessage> history = redisChatService.load(member.getMemberId(), product.getId());
+
+        // 3) 기존 DB 저장 기록 삭제
+        queryHistoryRepository.deleteByMemberAndProduct(member, product);
+
+        // 4) 저장 옵션이 켜져 있고 대화 내역이 있으면 db에 저장
+        if (!skipSave && !history.isEmpty()) {
+            List<QueryHistory> histories = new ArrayList<>(history.size());
+
+            for (ChatMessage message : history) {
+                QueryHistory qh = QueryHistory.builder()
+                    .product(product)
+                    .member(member)
+                    .role(convertRole(message.getRole()))
+                    .messages(message.getMessages())
+                    .build();
+                histories.add(qh);
+            }
+            queryHistoryRepository.saveAll(histories);
         }
 
-        return answer;
+        redisChatService.delete(member.getMemberId(), product.getId());
     }
 
+    // 문자열 role("user", "bot")을 ChatRole Enum 값으로 변환
+    private ChatRole convertRole(String role) {
+        if ("user".equals(role)) {
+            return ChatRole.user;
+        }
+
+        if ("bot".equals(role)) {
+            return ChatRole.bot;
+        }
+        throw new IllegalArgumentException("Unknown role: " + role);
+    }
```

Comment:
```suggestion
private ChatRole convertRole(String role) {
    return switch (role) {
        case "user" -> ChatRole.user;
        case "bot" -> ChatRole.bot;
        default -> throw new IllegalArgumentException("Unknown role: " + role);
    };
}
```
스위치문을 사용하면 코드가 더 간결해 보이네요
@yeon105